honeycombextension/extension.go

```diff
@@ -273,7 +272,6 @@ func (h *honeycombExtension) createUsageReport() ([]byte, error) {
 	}
 
 	// marshal the metrics into a byte slice
-	// TODO: if this marshal fails, we'll lose all the data we grabbed at the beginning of the function. Should we deal with that?
```

Comment:
Dropping the data is a feature - if we ever cant get the data out of the extension for any reason dropping it ensures we dont have runaway memory.